cmake/deps-helper.cmake

```diff
@@ -39,6 +39,20 @@ if(CMAKE_VERSION VERSION_GREATER_EQUAL "3.14")
     )
 endif()
 
+macro(check_auto_download name disable_option)
+    if(NOT ENABLE_AUTO_DOWNLOAD)
+        if("${disable_option}" STREQUAL "")
+            message(FATAL_ERROR "Could not find the required dependency ${name} \
+in the system. Please install it yourself or use --auto-download to let us \
+download and build it for you.")
+        else()
+            message(FATAL_ERROR "Could not find the optional dependency ${name} \
+in the system. You can disable this dependency with ${disable_option}, install \
+it yourself or use --auto-download to let us download and build it for you.")
+        endif()
+    endif()
+endmacro(check_auto_download)
```

Comment:
Should this specify the version that needs to be installed by the user?

Reply:
Good point. I'll try to print the required version, if there is one.